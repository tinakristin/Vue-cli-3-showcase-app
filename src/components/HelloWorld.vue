<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <hr/>
    <adresse-test></adresse-test>

  </div>
</template>

<script>
// import AdresseTest from './AdresseTest';

import AdresseTest from './AdresseTest';

export default {
  name: 'HelloWorld',
  components: { AdresseTest },
  props: {
    msg: {
      type: String,
      required: true,
    },
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  margin: 0 10px;
}
a {
  color: #42b983;
}

</style>
