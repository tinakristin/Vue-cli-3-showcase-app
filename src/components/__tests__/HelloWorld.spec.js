import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import HelloWorld from '../HelloWorld.vue';
import MockStore from '../../../tests/unit/mockVuexStore';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.component('font-awesome-icon', FontAwesomeIcon);

// Mocks
const store = new Vuex.Store(MockStore);

const propsData = {
  msg: 'Welcome',
};

describe('HelloWorld', () => {
  it('matches the snapshot', () => {
    const wrapper = shallowMount(HelloWorld, {
      localVue, store, propsData,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders props.msg when passed', () => {
    const msg = 'new message';
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }, localVue, store,
    });
    expect(wrapper.text()).toMatch(msg);
  });
});
