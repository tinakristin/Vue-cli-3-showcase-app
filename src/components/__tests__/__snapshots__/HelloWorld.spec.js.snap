// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld matches the snapshot 1`] = `
<div class="hello">
  <h1>Welcome</h1>
  <hr>
  <adresse-test-stub></adresse-test-stub>
</div>
`;
